### ISSUE TITLE
MonsterInsights 7.0.0 breaks plugin

### ISSUE THREAD

**Issue by dxdc** [jtsternberg/Google-Analytics-Top-Content-Widget#25]
This file (referenced here) is no longer available as of MonsterInsights 7.0.0, and `MonsterInsights_GA` class is also not present. Suspect this will take more than a few code changes, since they have modified their structure.

https://github.com/jtsternberg/Google-Analytics-Top-Content-Widget/blob/master/includes/class-ga-top-content.php#L668

- Maybe it would be helpful to include a soft fail or test if this file exists so it does not crash site upon future upgrade?

**Comment by jtsternberg** [comment]
Pinging @chriscct7 for some insight, as he helped address a few other similar issues, #18 and #23.

**Comment by jtsternberg** [comment]
Also see https://wordpress.org/support/topic/version-7-0-0-breaks-website/

**Comment by jtsternberg** [comment]
And https://wordpress.org/support/topic/not-support-google-analytics-for-wordpress-7-0-0/

**Comment by chriscct7** [comment]
@jtsternberg did you mean to close this?

There really isn't going to be an easy, if at all substitute. We forgot about this plugin and since no one was running stats through the old API we had to Google we were advised to shut it off so that we could move the quota to a different API. In MonsterInsights, we no longer directly query for data from Google, so there isn't a substitute for what the plugin was doing unfortunately at least that I can think of

**Comment by jtsternberg** [comment]
Oops, reopened. Ok, I was afraid of that. Hmm I will have to think about this. I have always wanted to avoid the entire GA connection UI that would be required w/o piggy-backing on the analytics plugin. I'll have to see if I can still use the GA connection you request/store, or if this is a bigger UI can of worms.

**Comment by chriscct7** [comment]
I put together a quick temp fix. I'd recommend doing a release with https://github.com/jtsternberg/Google-Analytics-Top-Content-Widget/pull/26 until something can be figured out that way at least there won't be sites going down with fatal errors.

**Comment by jtsternberg** [comment]
Ok, pushed out [a release](https://github.com/jtsternberg/Google-Analytics-Top-Content-Widget/releases/tag/v1.7.3), but just for those following along, this is not a fix, but instead disables analytics requests to keep your site from having fatal-errors. A permanent fix will need to be determined.

**Comment by dxdc** [comment]
thx for the quick response @jtsternberg ... hope we can still make use of the plugin despite the severe changes to the analytics!

@chriscct7 would it be possible to export the subsections of the Google query logic from the previous version or is that unmanageable?

**Comment by chriscct7** [comment]
Not really, I mean that's hundreds of files and thousands of lines of logic. You'd be basically recreating what was nearly the entire plugin

**Comment by manofmanytastes** [comment]
Hi Guys,

Thanks for all of your help on this so far @jtsternberg @chriscct7 . Just wondering if there's any update on a fix. The widget is still showing up as blank on our website manofmany.com

Cheers,
Scott

**Comment by AlessandroFerrariFood** [comment]
+1

**Comment by MartinHlavna** [comment]
Any updates? If there is something I can look at for help let me know.

**Comment by mtoensing** [comment]
Can`t we just ask the guys from monster insight for a special treatment? They could just bring back the methods for this plugin.

**Comment by chriscct7** [comment]
Hi there,
I don't have any updates on my side yet. We're trying to figure out something for this.

It's not as easy as just including an old function. We'd have to reincorporate literally many dozens to over a hundred files, plus thousands of lines of additional code, which would take us months to do and test and then we'd also have to get Google to reactivate the API used to process those requests which is a minimum of 6 weeks on their side to process the request.

We're trying to find a way of being able to funnel these requests through the new system but it's not something we've had luck with doing so far. They are completely different systems. I'm not optimistic in your chances on the method we're trying with right now, but we're doing our best to try to find a solution.

**Comment by MartinHlavna** [comment]
Hi,
any news on this?

**Comment by mtoensing** [comment]
It's dead, Jim.

**Comment by AlessandroFerrariFood** [comment]
Hi @jtsternberg,
I work for a small company and we were using your plugin for our website.
Do you think is it possible to fix/revive it if we can gather enough money to support the development?
Can you foresight a proper amount?
Anyone interested in this proposal?

**Comment by chriscct7** [comment]
As an update MonsterInsights is building our own first party add-on to do the features of this plugin plus some additional ones, and also have compatibility with Gutenberg. We don't have a precise eta but it will be released after November 19ths launch of WordPress with Gutenberg for sure as we want to make sure we are 100% compatible with it.

**Comment by AlessandroFerrariFood** [comment]
Thanks @chriscct7 
Let me know as soon as you have an eta.
The sooner the better :-)

**Comment by oburt** [comment]
We had held off on updating both the MonsterInsights plugin and Top Content Widget to keep this running as long as possible, but it looks like the OATH Client App that had been used for the 6.0 version of MonsterInsights has expired or been removed so now the Top Content Widget is fully dead for us.

So, @chriscct7, very much looking forward to hearing more about the upcoming MI add-on y'all are working on!

**Comment by ghost** [comment]
> We had held off on updating both the MonsterInsights plugin and Top Content Widget to keep this running as long as possible, but it looks like the OATH Client App that had been used for the 6.0 version of MonsterInsights has expired or been removed so now the Top Content Widget is fully dead for us.

We found a way to bypass the expired MonsterInsights Client App: There is a filter in the MonsterInsights plugin just for this, named "monsterinsights_lite_google_app_config".

First, we created an Google API authentication app. See [Analytics Reporting API - Authorization](https://developers.google.com/analytics/devguides/reporting/core/v4/authorization).
Then we created a plugin that looks like this (don’t forget to replace the **** place markers **** ):
```
<?php
/**
 * Plugin Name: **** a name **** 
 * Description: **** a description ****
 * Plugin URI:  **** a plugin URI ****
 * Author:      **** an author ****
 * Author URI:  **** an author URI ****
 * Version:     **** a version ****
 */

add_filter( 'monsterinsights_lite_google_app_config', 'your_function_name' ) ;

function your_function_name( $config ) {
	$config['application_name'] = '****  your app name ****' ;
	$config['client_id'] = '****  your client id ****' ;
	$config['client_secret'] = '****  your client secret ****' ;
	return $config ;
}
```
Finally, we activated the plugin and did the authentification with the new App.

Regards.

**Comment by oburt** [comment]
@technocyclope, that is a great find and idea! I tried setting up our own client authorization app but keep getting 403 accessNotConfigured errors. Could you provide any more details about what settings you used to create the replacement app? What the important option values are, etc?

**Comment by AlessandroFerrariFood** [comment]
@chriscct7
Gutenberg is out for a while.
Any news about MonsterInsights first party add-on?
We was very interested but if there's no news or realistic ETA we'll have to take other ways.
Thanks
